protocol/auditlog.go

```diff
@@ -112,26 +129,28 @@ func (l ConiksAuditLog) Insert(addr string, signKey sign.PublicKey,
 
 		// verify the consistency of each new STR before inserting
 		// into the audit log
-		if err := verifySTRConsistency(signKey, h.latestSTR, str); err != nil {
+		if err := h.verifySTRConsistency(snaps[i-1], str); err != nil {
 			return err
 		}
-
-		h.updateLatestSTR(snaps[i])
 	}
 
 	// Finally, add the new history to the log
+	if len(snaps) > 1 {
```

Comment:
This check only applies to `Insert`s because `snaps[0]` is inserted into `h.snapshots` in `newDirectoryHistory` and the rest of the range is added separately, while in all other cases (i.e. Update) we will always add the entire range at once. So moving this check into `updateVerifiedSTR` would have to distinguish between an`Insert` and an `Update`, which I think makes things messy.